sql/item_cmpfunc.cc

```diff
@@ -6212,8 +6212,8 @@ bool Regexp_processor_pcre::compile(String *pattern, bool send_error)
       int lmsg= pcre2_get_error_message(pcreErrorNumber,
                                         (PCRE2_UCHAR8 *)buff, sizeof(buff));
       if (lmsg >= 0)
-        my_snprintf(buff+lmsg, sizeof(buff)-lmsg,
-                    " at offset %d", pcreErrorOffset);
+        snprintf(buff+lmsg, sizeof(buff)-lmsg,
+                 " at offset %zu", pcreErrorOffset);
```

Comment:
`my_snprintf` could really use the `z`-size (for `size_t`) – this ain’t the only place that casts a `size_t` arg.

Reply:
Nevermind. `my_snprintf` supports `z`.
https://github.com/MariaDB/server/blob/25b5c63905682a0e38d8ed86170d98be6ddb8cf8/include/mysql/service_my_snprintf.h#L54

Reply:
Force-pushed a few amendments regarding `size_t` and [fixed-size integer types](https://en.cppreference.com/w/cpp/header/cinttypes#Macros)
Regarding the choice on `PRIu64` from C/C++ over `UINT64PF` from `univ.i`, see https://github.com/MariaDB/server/pull/3360#discussion_r1703530824.